src/caikit_nlp_client/grpc_client.py

```diff
@@ -159,32 +188,45 @@ def _close(self):
     def __del__(self):
         self._close()
 
-    def _make_channel(self, host: str, port: int, **kwargs) -> grpc.Channel:
-        log.debug(f"Making a channel for {host}:{port} with these kwargs={kwargs}")
-        if host.strip() == "":
+    def _make_channel(
+        self,
+        host: str,
+        port: int,
+        *,
+        insecure: bool = False,
+        ca_cert: Optional[bytes] = None,
+        client_key: Optional[bytes] = None,
+        client_cert: Optional[bytes] = None,
+        server_cert: Optional[bytes] = None,
+    ) -> grpc.Channel:
+        if not host.strip():
             raise ValueError("A non empty host name is required")
-        if port <= 0:
-            raise ValueError("A non zero port is required")
+        if int(port) <= 0:
+            raise ValueError("A non zero port number is required")
 
         connection = f"{host}:{port}"
-        ca_cert = kwargs.get("ca_cert")
+        if insecure:
+            log.warning("Connecting over an insecure grpc channel")
+            return grpc.insecure_channel(connection)
+
+        credentials_kwargs: dict[str, bytes] = {}
         if ca_cert:
-            log.debug("A CA certificate has been detected, creating a TLS channel")
-            return grpc.secure_channel(
-                connection,
-                grpc.ssl_channel_credentials(ca_cert),
-            )
-        client_key = kwargs.get("client_key")
-        client_cert = kwargs.get("client_cert")
-        server_cert = kwargs.get("server_cert")
+            log.info("Connecting using provided CA certificate for secure channel")
+            credentials_kwargs.update(root_certificates=ca_cert)
+
         if client_key and client_cert and server_cert:
-            log.debug(
-                "A client key, client and server certificates have been detected, \
-                creating a mTLS channel"
+            log.info("Connecting using mTLS for secure channel")
+            # TODO: check if server_cert has the same meaning as ca_cert, and get
```

Comment:
@dtrifiro, the code in the server is as follows:

```python
       if (
            self.tls_config
            and self.tls_config.server.key
            and self.tls_config.server.cert
        ):
            log.info("<RUN10001805I>", "Running with TLS")

            tls_server_pair = (
                bytes(self._load_secret(self.tls_config.server.key), "utf-8"),
                bytes(self._load_secret(self.tls_config.server.cert), "utf-8"),
            )
            if self.tls_config.client.cert:
                log.info("<RUN10001806I>", "Running with mutual TLS")
                # Combine the client cert with the server's own cert so that
                # health probes can use the server's key/cert instead of needing
                # one signed by a potentially-external CA.
                root_certificates = b"\n".join(
                    [
                        bytes(self._load_secret(self.tls_config.client.cert), "utf-8"),
                        tls_server_pair[1],
                    ]
                )
                # Client will verify the server using server cert and the server
                # will verify the client using client cert.
                server_credentials = grpc.ssl_server_credentials(
                    [tls_server_pair],
                    root_certificates=root_certificates,
                    require_client_auth=True,
                )
            else:
                server_credentials = grpc.ssl_server_credentials([tls_server_pair])
```